### PR TITLE
Adding docs related to MI secrets, cert and dropins

### PR DESCRIPTION
**Pull request by ruks**
## Purpose
- Fixes for
  - https://github.com/wso2-enterprise/choreo/issues/20993
  - https://github.com/wso2-enterprise/choreo/issues/21003
  - https://github.com/wso2-enterprise/choreo/issues/20815

### CODE REVIEW

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 1]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+
+Within the `dropins` directory, add the OSGi bundle JAR files. These bundles will be picked up by the OSGi runtime during
+deployment, enabling you to utilize their functionality within your Micro Integrator project.
 
 ![Libs directory](../assets/img/develop-components/micro-integrator/libs_dir.png)
 
+## Importing custom certificates to Micro Integrator
+
+The feature allows Choreo MI users to import certificates to the MI `client-truststore.jks` file. This is useful when
+users need to connect to servers or services that use SSL/TLS encryption and have self-signed or custom certificates
```
Comment (a5anka) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:82]:
```suggestion
users need to connect to servers or services that use SSL/TLS encryption and have self-signed or certificates issued by private CAs
```

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 2]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+root of your project. If your project root and GitHub repository root are not the same, place the `libs` directory in
+the project root. After adding the required JAR files to the `libs` directory, the Micro Integrator runtime will
+automatically include them when deploying the component. This process ensures that any dependencies needed by your
+integration flow are available at runtime.
+
+In addition to `jar` libraries, you may also need to incorporate OSGi bundle JARs into your project. To do this,
+create another directory called `dropins` at the root of your project. Similarly, if the project root and GitHub
+repository root differ, place the `dropins` directory in the project root.
+
+Within the `dropins` directory, add the OSGi bundle JAR files. These bundles will be picked up by the OSGi runtime during
```
Comment (shaniR) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:74]: Let's not make this a separate paragraph, and have it in the same para.

Comment (shaniR) [reply en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:74]: These bundles will be picked up by the OSGi runtime during -> The OSGI runtime will pick up these bundles

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 3]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+deployment, enabling you to utilize their functionality within your Micro Integrator project.
 
 ![Libs directory](../assets/img/develop-components/micro-integrator/libs_dir.png)
 
+## Importing custom certificates to Micro Integrator
```
Comment (shaniR) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:79]: to MI

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 4]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+
+The feature allows Choreo MI users to import certificates to the MI `client-truststore.jks` file. This is useful when
```
Comment (shaniR) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:81]: Let's rephrase it as :
The feature enables Choreo MI users to import certificates into the MI `client-truststore.jks` file. This functionality proves valuable when users need to establish connections with servers or services utilizing SSL/TLS encryption and possessing self-signed certificates or certificates issued by private certificate authorities (CAs) that are not automatically trusted.

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 5]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+users need to connect to servers or services that use SSL/TLS encryption and have self-signed or certificates issued by private CAs
+that are not trusted by default. By importing the certificate to the MI client-truststore.jks file, MI clients can
```
Comment (shaniR) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:83]: Rephrase as "Importing the certificate to the MI `client-truststore.jks` file enables MI clients to establish secure connections with the server or service seamlessly, eliminating SSL/TLS errors or warnings."

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 6]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+establish a secure connection with the server or service without encountering SSL/TLS errors or warnings.
+
+Before importing the certificate, you need to obtain the certificate file. This file can be in
```
Comment (shaniR) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:86]: Rephrase as : 
Before importing the certificate, it is necessary to obtain the certificate file, which can be in PEM or DER format. You can get the certificate from the server or service provider or export it from a web browser. Certificates should be added to the /wso2mi/certs/ folder. Users need to verify that the mount path for the certificate file aligns with this specific path.

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 7]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+the PEM or DER format. You can obtain the certificate file from the server or service provider or by exporting it from a
+web browser. Certificates should be added to the `/wso2mi/certs/` folder. Therefore, users must ensure that the mount path
+for the certificate file follows this path
+
+1. Go to the Component page and click on the **Deploy** page for the component you want to add the certificate to.
```
Comment (shaniR) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:91]: 1. In the Choreo console, select the component you wish to add a certificate to. 
2. From the left navigation click **Deploy**.

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 8]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+
+## Working with sensitive data using MI Secure Vault
+
+MI Secure Vault is a feature that allows users to securely store sensitive data, such as passwords and tokens, and
+provide access to that data to MI Integrations. This feature provides an added layer of security to your integrations by
+reducing the risk of accidental exposure of sensitive data.
+
+To use MI Secure Vault to work with sensitive data, follow these steps:
+
+1. Go to the Component page and click on the **Deploy** page.
```
Comment (shaniR) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:114]: 1. Select the component from the Choreo Console. 
2. From the left navigation, click **Deploy**.

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md [hunk 9]
```diff
@@ -61,10 +61,83 @@ Here are some of the benefits of adding third-party JARs to the Micro Integrator
...
+2. Click on the **Configure and Deploy**.
+3. Click on the **Add Environment Configurations**
+4. Specify the following values as configurations:
+
+    | **Field**       | **Value**                                 |                                    
+    |-------------------------------------------|--------------------------------------------------------|
+    | **Name** | An appropriate name for the secret/alias. |
+    | **Value**       | Value for the secret                      |
+
+5. Click **Mark as a Secret**.
```
Comment (shaniR) [inline en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:124]: Select **Mark as a secret**

**Review by shaniR — CHANGES_REQUESTED** [review]

File: en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md (hunk 2 above)
Comment (ruks) [reply en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:74]: you mean
```
The OSGI runtime will pick up these bundles, enabling you to utilize their
functionality within your Micro Integrator project.
```

Comment (shaniR) [reply en/docs/develop-components/work-with-the-micro-integrator-runtime-in-choreo.md:74]: The OSGI runtime will pick up these bundles during deployment, enabling you to utilize their functionality within your Micro Integrator project.

**Review by shaniR — CHANGES_REQUESTED** [review]

**Review by shaniR — APPROVED** [review]

**Pull request merged**